src/main/java/io/tarantool/driver/core/SingleValueCallResultImpl.java

```diff
@@ -20,34 +23,41 @@
     private final T value;
 
     public SingleValueCallResultImpl(ArrayValue result, ValueConverter<Value, T> valueConverter) {
-        if (result == null) {
-            throw new TarantoolFunctionCallException("Function call result is null");
-        }
-        if (result.size() == 0 || result.size() == 1 && result.get(0).isNilValue()) {
-            // [nil] or []
-            value = null;
-        } else if (result.size() == 2 && (result.get(0).isNilValue() && !result.get(1).isNilValue())) {
-            // [nil, "Error msg..."] or [nil, {str="Error msg...", stack="..."}]
-            throw TarantoolErrorsParser.parse(result.get(1));
-        } else {
-            // [result]
-            value = valueConverter.fromValue(result.get(0));
-        }
+        value = parseResult(result, valueConverter::fromValue);
     }
 
     public SingleValueCallResultImpl(ArrayValue result, MessagePackValueMapper valueMapper) {
+        value = parseResult(result, valueMapper::fromValue);
+    }
+
+    private T parseResult(ArrayValue result, Function<Value, T> valueGetter) {
         if (result == null) {
             throw new TarantoolFunctionCallException("Function call result is null");
         }
-        if (result.size() == 0 || result.size() == 1 && result.get(0).isNilValue()) {
+        int callResultSize = result.size();
+        Value resultValue = result.getOrNilValue(0);
+        Value errorsValue = result.getOrNilValue(1);
+        if (callResultSize == 0 || result.size() == 1 && resultValue.isNilValue()) {
             // [nil] or []
-            value = null;
-        } else if (result.size() == 2 && (result.get(0).isNilValue() && !result.get(1).isNilValue())) {
+            return null;
+        } else if (callResultSize == 2 && (resultValue.isNilValue() && !errorsValue.isNilValue())) {
             // [nil, "Error msg..."] or [nil, {str="Error msg...", stack="..."}]
-            throw TarantoolErrorsParser.parse(result.get(1));
+            throw TarantoolErrorsParser.parse(errorsValue);
+        } else if (callResultSize == 2 && errorsValue.isArrayValue()) {
+            // [result, nil] or [result, errors]
```

Comment:
You checked that second argument is not nil
```suggestion
            // [result, []] or [result, errors]
```

Reply:
Fixed the comment, thanks